### PR TITLE
Handle lengthy and newline character text

### PR DESCRIPTION
**Pull request by rajsync**
this is related to https://github.com/mapgears/ol3-google-maps/issues/323 issue.
Modified MapLabel.js to handle lengthy text, default canvas created has default width and height. lengthy text with new line character width and height is calculated and canvas width and height is updated.

canvas fill text function doesn't handle newline, hence its handles by calculating text line height and calling multiple fill text

### CODE REVIEW

File: src/olgm/gm/MapLabel.js [hunk 1]
```diff
@@ -127,11 +127,45 @@ class MapLabel extends MapElement {
         ctx.lineWidth = strokeWeight;
         ctx.strokeText(text, x, y);
       }
+      this.wrapText(ctx, text, x, y);
+      // ctx.fillText(text, x, y);
```
Comment (adube) [inline src/olgm/gm/MapLabel.js]: Remove the above line that's left commented out

File: src/olgm/gm/MapLabel.js [hunk 2]
```diff
@@ -127,11 +127,45 @@ class MapLabel extends MapElement {
...
+    }
+  }
 
-      ctx.fillText(text, x, y);
+  measureCanvas() {
+    var words = this.get('text').split('\r\n'),
```
Comment (adube) [inline src/olgm/gm/MapLabel.js]: Use `const` and `let` instead of `var` everywhere.

File: src/olgm/gm/MapLabel.js [hunk 3]
```diff
@@ -127,11 +127,45 @@ class MapLabel extends MapElement {
...
+        this.canvas_.width += metrics.width;
+      }
     }
+    if (lineHeight > this.canvas_.height) {
+      this.canvas_.height = y;
+    }
+    this.canvas_.width *= 2;
   }
 
+  wrapText(context, text, x, y) {
```
Comment (adube) [inline src/olgm/gm/MapLabel.js:191]: Add comment block for method, including the typing of each arguments here and their documentation.

**Review by adube — CHANGES_REQUESTED** [review]
Thanks for the patch.  I've got a few minor adjustments I'd like you to make.

Also, it would be nice if we could see the fix in action through an example.  Maybe add an additional feature to the "label.js" example that shows how the patch fixes the issue?

Thank you very much and happy coding.

**Comment by adube** [comment]
@rajsync Oh,I forgot to ask you: after this patch is merged in master, do you need to have a new npm release of olgm being made?  Thanks.

**Comment by rajsync** [comment]
> @rajsync Oh,I forgot to ask you: after this patch is merged in master, do you need to have a new npm release of olgm being made? Thanks.

@adube Yes, Please do a npm release. Thanks.

**Review by adube — CHANGES_REQUESTED** [review]
Thanks for the fixes.

Once last detail: would you please run `npm run lint` and fix all issues reported by the linter?  Then we'll be good to go for a merge / release.

Thank you.

**Comment by rajsync** [comment]
> Thanks for the fixes.
> 
> Once last detail: would you please run `npm run lint` and fix all issues reported by the linter? Then we'll be good to go for a merge / release.
> 
> Thank you.

@adube lint issues are fixed,thanks.

**Review by adube — APPROVED** [review]
Everything's clean.  I've checked the example once again and everything's behaving correctly.

Thanks again for the contribution.  That's really really appreciated.

I'll take care of merging it.

**Comment by adube** [comment]
@rajsync OLGM v1.7.1 is released on npm.  Thanks for the patch and happy coding.

**Comment by rajsync** [comment]
> @rajsync OLGM v1.7.1 is released on npm. Thanks for the patch and happy coding.

thanks and appreciate fast response and release @adube

**Pull request merged**